etl/src/energuide/embedded/basement.py

```diff
@@ -141,19 +158,28 @@ def _from_data(cls, floor: element.Element, construction_type: str, floor_type:
         )
 
     @classmethod
-    def from_basement(cls, floor: element.Element) -> typing.List['BasementFloor']:
-        return [cls._from_data(floor, 'AddedToSlab', FloorType.SLAB)]
+    def from_basement(cls, floor: typing.Optional[element.Element]) -> typing.List['BasementFloor']:
+        return [
+            cls._from_data(floor, 'AddedToSlab', FloorType.SLAB)
+            if floor is not None else cls._empty_floor(FloorType.SLAB)
+        ]
 
     @classmethod
-    def from_crawlspace(cls, floor: element.Element) -> typing.List['BasementFloor']:
+    def from_crawlspace(cls, floor: typing.Optional[element.Element]) -> typing.List['BasementFloor']:
         return [
-            cls._from_data(floor, 'AddedToSlab', FloorType.SLAB),
-            cls._from_data(floor, 'FloorsAbove', FloorType.FLOOR_ABOVE_CRAWLSPACE),
+            cls._from_data(floor, 'AddedToSlab', FloorType.SLAB)
+            if floor is not None else cls._empty_floor(FloorType.SLAB),
+
+            cls._from_data(floor, 'FloorsAbove', FloorType.FLOOR_ABOVE_CRAWLSPACE)
+            if floor is not None else cls._empty_floor(FloorType.FLOOR_ABOVE_CRAWLSPACE),
```

Comment:
I think this would be easier to read if you reversed the logic:
```
if floor is None:
    return [cls._empty_floor(FloorType.SLAB), cls._empty_floor(FloorType.FLOOR_ABOVE_CRAWLSPACE)]
else:
    return [
        cls._from_data(floor, 'AddedToSlab', FloorType.SLAB),
        cls._from_data(floor, 'FloorsAbove', FloorType.FLOOR_ABOVE_CRAWLSPACE),
    ]
```